src/Uno.UI/UI/Xaml/Controls/TextBox/OverlayTextBoxView/OverlayTextBoxViewExtension.skia.cs

```diff
@@ -0,0 +1,290 @@
+	{
+		_owner = owner ?? throw new ArgumentNullException(nameof(owner));
+		_textBoxViewFactory = textBoxViewFactory ?? throw new ArgumentNullException(nameof(textBoxViewFactory));
+	}
+
+	public abstract bool IsOverlayLayerInitialized(XamlRoot xamlRoot);
+
+	public void StartEntry()
+	{
+		if (_owner.TextBox is not { } textBox ||
+			_owner.TextBox.XamlRoot is not { } xamlRoot)
+		{
+			// The parent TextBox must exist as source of properties.
+			return;
+		}
+
+		_contentElement = textBox.ContentElement;
+
+		EnsureTextBoxView(textBox);
+		ObserveNativeTextChanges();
+		_lastSize = new Size(-1, -1);
+		_lastPosition = new Point(-1, -1);
+		UpdateNativeView();
+		SetNativeText(textBox.Text);
+		InvalidateLayout();
+
+		_textBoxView!.AddToTextInputLayer(xamlRoot);		
+		_textBoxView.SetFocus(true);
+
+		// Selection is now handled by native control
+		if (_selectionStartCache != null && _selectionLengthCache != null)
+		{
+			Select(_selectionStartCache.Value, _selectionLengthCache.Value);
+		}
+		else
+		{
+			// Select end of the text
+			var endIndex = textBox.Text.Length;
+			Select(endIndex, 0);
+		}
+		_selectionStartCache = null;
+		_selectionLengthCache = null;
+	}
+
+	public void EndEntry()
+	{
+		_textChangedDisposable.Disposable = null;
+		if (_textBoxView is null ||
+			!_textBoxView.IsDisplayed)
+		{
+			// No entry in progress
+			return;
+		}
+
+		if (GetNativeText() is { } inputText)
+		{
+			_owner.UpdateTextFromNative(inputText);
+		}
+
+		_contentElement = null;
+
+		if (_textBoxView != null)
+		{
+			var bounds = GetNativeSelectionBounds();
+			(_selectionStartCache, _selectionLengthCache) = (bounds.start, bounds.end - bounds.start);
+			_textBoxView.RemoveFromTextInputLayer();
+		}
+	}
+
+	public void SetText(string text) => SetNativeText(text);
+
+	public void UpdateNativeView()
+	{
+		if (_textBoxView is null || _owner.TextBox is not { } textBox)
+		{
+			// If the input widget does not exist, we don't need to update it.
+			// The parent TextBox must exist as source of properties.
+			return;
+		}
+
+		EnsureTextBoxView(textBox);
+	}
+
+	public void InvalidateLayout()
+	{
+		UpdateSize();
+		UpdatePosition();
+	}
+
+	public void UpdateProperties()
+	{
+		if (_owner?.TextBox is { } textBox)
+		{
+			_textBoxView?.UpdateProperties(textBox);
+		}
+	}
+
+	public void UpdateSize()
+	{
+		if (_contentElement is null ||
+			_textBoxView is null ||
+			!_textBoxView.IsDisplayed)
+		{
+			return;
+		}
+
+		var width = (int)(_contentElement.ActualWidth - _contentElement.Padding.Horizontal());
+		var height = (int)(_contentElement.ActualHeight - _contentElement.Padding.Vertical());
+
+		if (_lastSize.Width != width || _lastSize.Height != height)
+		{
+			_lastSize = new Size(width, height);
+			_textBoxView.SetSize(_lastSize.Width, _lastSize.Height);
+		}
+	}
+
+	public void UpdatePosition()
+	{
+		if (_contentElement == null ||
+			_textBoxView == null ||
+			!_textBoxView.IsDisplayed)
```

Comment:
Minor stylistic suggestion

```suggestion
		if (_contentElement == null || _textBoxView is not { IsDisplayed: true })
```